### PR TITLE
feature: S3C-4342 Get Account and Test Cases

### PR DESCRIPTION
**Pull request by saitharunthodupunuru**

### CODE REVIEW

File: vaultclient/src/main/java/com/scality/vaultclient/dto/GetAccountRequestMarshaller.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        super.marshall(getAccountRequestDTO, protocolMarshaller, GET_ACCOUNT_ACTION);
+
+        try {
+            if (StringUtils.isNullOrEmpty(getAccountRequestDTO.getAccountName())
+                    && StringUtils.isNullOrEmpty(getAccountRequestDTO.getAccountId())
+                    && StringUtils.isNullOrEmpty(getAccountRequestDTO.getAccountArn())
+                    && StringUtils.isNullOrEmpty(getAccountRequestDTO.getCanonicalId())
+                    && StringUtils.isNullOrEmpty(getAccountRequestDTO.getEmailAddress()))
+             {
+                throw new VaultClientException("At least one of account-name, account-id, email, account-arn or canonical-id needs to be specified");
```
Comment (anurag4DSB) [inline vaultclient/src/main/java/com/scality/vaultclient/dto/GetAccountRequestMarshaller.java:53]: As per the code changes in vault only 1 should be specified.
Reference: https://github.com/scality/Vault/blob/development/7.10/lib/Validator/getAccount.js#L27

**Review by anurag4DSB — CHANGES_REQUESTED** [review]
getAccount parameters are exclusive, might need some changes.
Reference:
* Vault: https://github.com/scality/Vault/blob/development/7.10/lib/Validator/getAccount.js#L27

* Vaultclient: https://github.com/scality/vaultclient/blob/development/8.2/lib/IAMClient.js#L444

File: vaultclient/src/main/java/com/scality/vaultclient/dto/GetAccountRequestMarshaller.java (hunk 1 above)
Comment (saitharunthodupunuru) [reply vaultclient/src/main/java/com/scality/vaultclient/dto/GetAccountRequestMarshaller.java:53]: Correct. As this is only a client I am not validating if multiple parameters are provided. This is only a validation to verify if no parameter is passed. This is based on `vaultclient` code
Reference: https://github.com/scality/vaultclient/blob/7205377ec11aed8e682017ab7cb9d066eaf8cd90/lib/IAMClient.js#L407

**Review by anurag4DSB — APPROVED** [review]

File: vaultclient/src/main/java/com/scality/vaultclient/dto/GetAccountRequestMarshaller.java (hunk 1 above)
Comment (anurag4DSB) [reply vaultclient/src/main/java/com/scality/vaultclient/dto/GetAccountRequestMarshaller.java:53]: I would suggest, eventually, we need to validate if multiple parameters are provided. Would definitely help save some network expenses.
If you are not doing it now, worth creating a ticket for it.

This is how we do it in Vaultclient: https://github.com/scality/vaultclient/blob/development/7.10/lib/IAMClient.js#L444

Comment (saitharunthodupunuru) [reply vaultclient/src/main/java/com/scality/vaultclient/dto/GetAccountRequestMarshaller.java]: @anurag4DSB  Fixed in 98ba3dcae6bf39c87201fb20481e7cad1f79e1f2

**Review by anurag4DSB — APPROVED** [review]

File: vaultclient/src/test/java/com/scality/vaultclient/services/GetAccountServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+            getAccountMockClient.getAccount(getAccountRequestDTO).getAwsResponse();
+        }, "Expected SdkClientException");
+    }
+
+    /** Tests with Actual Vault **/
+    @Disabled
+    @Test
+    @SuppressWarnings( "deprecation" )
+    public void testGetAccountWithActualVault() {
+        //"D4IT2AWSB588GO5J9T00": "UEEu8tYlsOGGrgf4DAiSZD6apVNPUWqRiPG0nTB6"
```
Comment (anurag4DSB) [inline vaultclient/src/test/java/com/scality/vaultclient/services/GetAccountServiceTest.java:222]: Maybe add another comment that these are the default admin keys for vault.
Along with a link to the file: https://github.com/scality/Vault/blob/development/7.10/tests/utils/admincredentials.json

**Review by anurag4DSB — APPROVED** [review]

**Review by tmacro — APPROVED** [review]

**Comment by saitharunthodupunuru** [comment]
squashed and rebased review commits

**Pull request merged**